test/ntp_client_test.rb

```diff
@@ -0,0 +1,487 @@
+        .with(["/etc/ntp.conf"]).and_return(true)
+      allow(subject).to receive(:ProcessNtpConf)
+      allow(subject).to receive(:ReadSynchronization)
+      allow(Yast::SuSEFirewall).to receive(:Read)
+      allow(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".sysconfig.ntp.NTPD_RUN_CHROOTED"))
+        .and_return(nil)
+      allow(Yast::Service).to receive(:Enabled).with("ntpd").and_return(true)
+    end
+
+    it "returns true if config was read previously" do
+      subject.config_has_been_read = true
+      expect(Yast::Mode).not_to receive(:normal)
+
+      expect(subject.Read).to eql(true)
+    end
+
+    it "returns false if abort is pressed" do
+      allow(subject).to receive(:Abort).and_return(true)
+
+      expect(subject.Read).to eql(false)
+    end
+
+    it "doesn't show progress if it is not in normal Mode" do
+      expect(Yast::Progress).not_to receive(:New)
+
+      subject.Read
+    end
+
+    it "reads network interfaces config" do
+      expect(Yast::NetworkInterfaces).to receive(:Read)
+
+      subject.Read
+    end
+
+    it "reads ntp policy setting, using auto as default if not exists" do
+      expect(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".sysconfig.network.config.NETCONFIG_NTP_POLICY"))
+        .and_return(nil)
+      expect(subject.ntp_policy).to eql("auto")
+
+      subject.Read
+    end
+
+    it "loads known ntp servers and known country names" do
+      expect(subject).to receive(:GetNtpServers)
+      expect(subject).to receive(:GetCountryNames)
+
+      subject.Read
+    end
+
+    it "returns false if the ntp package is not and could not be installed" do
+      expect(Yast::PackageSystem).to receive(:CheckAndInstallPackagesInteractive)
+        .with(["ntp"]).and_return(false)
+      expect(Yast::Service).not_to receive(:Enabled)
+
+      expect(subject.Read).to eql(false)
+    end
+
+    it "checks if ntpd service is enable" do
+      expect(Yast::Service).to receive(:Enabled).with("ntpd")
+
+      subject.Read
+    end
+
+    it "checks active directory's ntp dumped data file" do
+      expect(Yast::FileUtils).to receive(:Exists).with("#{Yast::Directory.vardir}/ad_ntp_data.ycp")
+
+      subject.Read
+    end
+
+    it "reads ntp config from /etc/ntp.conf" do
+      expect(subject).to receive(:ProcessNtpConf)
+
+      subject.Read
+    end
+
+    it "reads synchronization config" do
+      expect(subject).to receive(:ReadSynchronization)
+
+      subject.Read
+    end
+
+    it "reads ntpd chroot configuration" do
+      expect(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".sysconfig.ntp.NTPD_RUN_CHROOTED"))
+
+      subject.Read
+    end
+
+    it "returns true if all reads were performed" do
+      expect(subject.Read).to eql(true)
+    end
+  end
+
+  describe "#Write" do
+    let(:data_dir) { File.join(File.dirname(__FILE__), "data") }
+
+    around do |example|
+      ::FileUtils.cp(File.join(data_dir, "scr_root/etc/ntp.conf.original"),
```

Comment:
Maybe it's just a matter of taste, but I prefer to not modify files which are under version control. I would copy the whole directory to a temporal one to perform tests. The reason is that you could modify those files during a test run and commit them by accident.

And what if you have three or four configuration files?